### PR TITLE
Test commit of HIP initiated linking

### PR DESCRIPTION
**Pull request by VenuAjitesh**
The implementation i have done, and the flow goes like :

1) facade gets request from Legacy about linkingRecords
2) facade Hip initiates the linking flow and responds legacy with status code of init api.
3) using the requestId, the legacy can track the status ie: Initiated or Success.
collection-MongoDb

### CODE REVIEW

File: src/main/java/com/wrapper/Controller/ResponseController/ResponseData.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+			this.hipGateway.startConfirmCall(data);
+		} else {
+			log.info("Error in onInitCall: " + data.getError());
+		}
+
+	}
+
+	@PostMapping({"/v0.5/users/auth/on-confirm"})
+	public String onConfirmCall(@RequestBody OnConfirmResponse data) throws IOException, URISyntaxException {
+		if (data.getError() == null) {
```
Comment (atulai-sg) [inline src/main/java/com/wrapper/Controller/ResponseController/ResponseData.java:54]: null check for data as well..
data != null && data.getError() == null

File: src/main/java/com/wrapper/Model/CareContextTable.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
+//
+// Source code recreated from a .class file by IntelliJ IDEA
+// (powered by FernFlower decompiler)
+//
+
+package com.wrapper.Model;
```
Comment (atulai-sg) [inline src/main/java/com/wrapper/Model/CareContextTable.java:6]: lets create a package named mongo.tables and have these classes be created under it

File: src/main/java/com/wrapper/Model/CareContextTable.java [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+import com.wrapper.ResponseController.LinkRecordsResponse;
+import java.util.List;
+import org.springframework.data.mongodb.core.index.Indexed;
+import org.springframework.data.mongodb.core.mapping.Document;
+import org.springframework.data.mongodb.core.mapping.Field;
+
+@Document(
+		collection = "careContexts"
+)
+public class CareContextTable {
```
Comment (atulai-sg) [inline src/main/java/com/wrapper/Model/CareContextTable.java:17]: Lets rename this to CareContext

File: src/main/java/com/wrapper/Model/CareContextTable.java [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+	@Field("initRequestId")
+	@Indexed(
+			unique = true
+	)
+	public String initRequestId;
+	@Field("confirmRequestId")
```
Comment (atulai-sg) [inline src/main/java/com/wrapper/Model/CareContextTable.java:23]: lets give breakline after each variable

File: src/main/java/com/wrapper/Model/CareContextTable.java [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+	@Indexed(
+			unique = true
+	)
```
Comment (atulai-sg) [inline src/main/java/com/wrapper/Model/CareContextTable.java:26]: I am hoping that indentation and other formatting issues will be fixed once you implement spotless plugin

File: src/main/java/com/wrapper/RequestBody/AuthInitBody.java [hunk 6]
```diff
@@ -0,0 +1,70 @@
+//
+// Source code recreated from a .class file by IntelliJ IDEA
+// (powered by FernFlower decompiler)
+//
+
+package com.wrapper.RequestBody;
```
Comment (atulai-sg) [inline src/main/java/com/wrapper/RequestBody/AuthInitBody.java:6]: lets call this package as requests

File: src/main/java/com/wrapper/ResponseController/OnConfirmResponse.java [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+	public RespData getResp() {
+		return this.resp;
+	}
+
+	public void setResp(RespData resp) {
+		this.resp = resp;
+	}
+
+	public String printData() {
+		String var10000 = this.getResp().getRequestId();
```
Comment (atulai-sg) [inline src/main/java/com/wrapper/ResponseController/OnConfirmResponse.java:62]: why are we choosing variable name as such?

File: src/main/java/com/wrapper/ResponseController/OnConfirmResponse.java [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+	public RespData getResp() {
+		return this.resp;
+	}
+
+	public void setResp(RespData resp) {
+		this.resp = resp;
+	}
+
+	public String printData() {
```
Comment (atulai-sg) [inline src/main/java/com/wrapper/ResponseController/OnConfirmResponse.java:61]: lets just log and not print any thing..we can set log level as info and debug but that can be done later as well and not on priority

File: src/main/java/com/wrapper/RequestBody/AddCareContextBody.java [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+		}
+
+		patientNode.set("careContexts", careContextArray);
+		linkNode.set("patient", patientNode);
+		requestBody.set("link", linkNode);
+		HttpEntity<ObjectNode> requestEntity = new HttpEntity(requestBody, this.commonUtility.getHeaders());
+		log.info(requestEntity.getHeaders());
+		log.info(requestEntity.getBody());
+		return requestEntity;
+	}
```
Comment (atulai-sg) [inline src/main/java/com/wrapper/RequestBody/AddCareContextBody.java:78]: I am unsure what is happening here. This request body should be plain simple POJO and caller should form this request body with appropriate values

File: src/main/java/com/wrapper/RequestBody/AddCareContextBody.java [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+import java.util.UUID;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.bind.annotation.RequestBody;
+
+@Component
+public class AddCareContextBody {
```
Comment (atulai-sg) [inline src/main/java/com/wrapper/RequestBody/AddCareContextBody.java:27]: lets call this as CareContextRequest

File: src/main/java/com/wrapper/ResponseController/LinkRecordsResponse.java [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+// Source code recreated from a .class file by IntelliJ IDEA
+// (powered by FernFlower decompiler)
+//
+
+package com.wrapper.ResponseController;
+
+import java.io.Serializable;
+import java.util.List;
+
+public class LinkRecordsResponse implements Serializable {
```
Comment (atulai-sg) [inline src/main/java/com/wrapper/ResponseController/LinkRecordsResponse.java:11]: lets use lombok's Data annotation for getters and setters

**Review by atulai-sg — COMMENTED** [review]
Thanks @VenuAjitesh for this pull request. This looks good but we would need certain changes to give this a good shape.

File: src/main/java/com/wrapper/Controller/HRP/HipGateway.java [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.client.RestTemplate;
+
+@RestController
+public class HipGateway {
+	private static final Logger log = LogManager.getLogger(HipGateway.class);
+	RestTemplate restTemplate;
```
Comment (MJ1998) [inline src/main/java/com/wrapper/Controller/HRP/HipGateway.java:29]: Are we using this ?

File: src/main/java/com/wrapper/Services/ConfirmAuth.java [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+
+package com.wrapper.Services;
+
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.wrapper.ResponseController.OnInitResponse;
+import java.io.FileNotFoundException;
+import java.net.URISyntaxException;
+import org.springframework.http.ResponseEntity;
+
+public interface ConfirmAuth {
```
Comment (MJ1998) [inline src/main/java/com/wrapper/Services/ConfirmAuth.java:14]: Appending Service at the end of the class name helps!

File: src/main/java/com/wrapper/Controller/HRP/HipGateway.java [hunk 14]
```diff
@@ -0,0 +1,81 @@
...
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.client.RestTemplate;
+
+@RestController
+public class HipGateway {
```
Comment (MJ1998) [inline src/main/java/com/wrapper/Controller/HRP/HipGateway.java:27]: I dont think this is a controller. 

Probably another service layer that the HIP Facade uses.

**Review by MJ1998 — CHANGES_REQUESTED** [review]
Thanks @VenuAjitesh for such fast work.

This is understandable but there are some points that can improve code health:-
1. Can you add java doc for controllers and important flows.
2. Append / Prepend the pattern type to each class name, for example, for each Service add "Service" to the class name like ConfirmAuthService.
3. Maintain [java naming conventions](https://www.thoughtco.com/using-java-naming-conventions-2034199)

**Pull request closed without merging**